api/v1/agentconfig_types.go

```diff
@@ -70,8 +71,11 @@ type AgentConfigSpec struct {
 	// This can be useful for a bundle which is targeting the kubernetes cluster that the operator is installed in.
 	// +optional
 	InstallationServiceAccount string `json:"installationServiceAccount,omitempty" mapstructure:"installationServiceAccount,omitempty"`
+
+	// PluginConfigFile specifies plugins required to run Porter bundles.
+	// In order to utilize mapstructure omitempty tag with an embedded struct, this field needs to be a pointer
 	// +optional
-	Plugins map[string]Plugin `json:"plugins,omitempty" mapstructure:"plugins,omitempty"`
+	PluginConfigFile *PluginFileSpec `json:"pluginConfigFile,omitempty" mapstructure:"pluginsConfigFile,omitempty"`
```

Comment:
You are using both `plugins` plural and singular here in the json/mapsturcture tags. I'm not sure which one you want to use?